NEWS.md

```diff
@@ -44,130 +45,133 @@
-  This affects some code that uses positional arguments. (#935, @michaelchirico)
+  This affects some code that uses positional arguments. (#935, @MichaelChirico)
   + For `lint()`, `...` is now the 3rd argument, where earlier this was `cache`
   + For `lint_dir()` and `lint_package()`, `...` is now the 2nd argument, where earlier this was `relative_path`
 * Argument `source_file` to exported functions `with_id()` and `ids_with_token()` has been renamed to
   `source_expression` to better reflect that this argument is typically the output of `get_source_expressions()`.
-  It has also been renamed as the argument of the now-private functional versions of many linters, which has no direct
-  effect on packages importing lintr, but is mentioned in case custom linters imitating `lintr` style have also
-  adopted the `source_file` naming and want to adapt to keep in sync.
+  For now, `source_file` can still be used (with warning). It has also been renamed as the argument of the now-private
+  functional versions of many linters, which has no direct effect on packages importing lintr, but is mentioned in case
+  custom linters imitating `lintr` style have also adopted the `source_file` naming and want to adapt to keep in sync.
 * Deprecated `with_defaults()` in favor of `linters_with_defaults()`, and add `modify_defaults()` which is intended to be used
   more generally to modify (i.e., extend, trim, and/or update) a list of defaults. Note that the argument corresponding to
   `with_defaults()`'s `default=` is called `defaults=` (i.e., pluralized) in both of these, and that usage like
   `with_defaults(default = NULL, ...)` should be converted to `linters_with_defaults(defaults = list(), ...)`
-  (#1029, #1336, #1361, @AshesITR and @michaelchirico)
+  (#1029, #1336, #1361, @AshesITR and @MichaelChirico)
 
 ## Other changes to defaults
 
 ### Updates to `default_linters`
 
 * New `brace_linter()` which combines several curly brace related linters, deprecating the following predecessors (#1041, @AshesITR):
-   + `closed_curly_linter()`; both now also allow `}]` in addition to `})` and `},` as exceptions.
+   + `closed_curly_linter()`; both now also allow `}]` in addition to `})` and `},` as exceptions, i.e., `}` doesn't need
+     to be on its own line if paired with a closing square bracket, a closing parenthesis, or a comma. Also improved lint metadata
+     so that source markers land at the opening brace instead of the closing parenthesis to improve the experience of fixing the lint (#583, @AshesITR)
    + `open_curly_linter()`; both also no longer lint unnecessary trailing whitespace (use `trailing_whitespace_linter()` for this)
-     and also allow `(`, `,`, and `%>%` on preceding lines as exceptions. (#487, #1028)
+     and also allow `(`, `,`, and `%>%` on preceding lines as exceptions, i.e., `{` can be alone on a line if the previous line
+     is terminated with an opening parenthesis, a comma, or a pipe (`%>%`). (#487, #1028, @AshesITR)
```

Comment:
`%>%` / `|>`?

Reply:
not `|>`, which doesn't support this syntactically (and `PIPE` is not in the XPath)

Reply:
Oh, right. Thanks!